docs/articles/viewing-epoch-based-metrics.rst

```diff
@@ -0,0 +1,167 @@
+install the Determined library and start a cluster locally.
+
+-  Ensure you have Docker running and then run the following command:
+
+.. code::
+
+   pip install determined
+
+   # If your machine has GPUs:
+   det deploy local cluster-up
+
+   # If your machine does not have GPUs:
+   det deploy local cluster-up --no-gpu
+
+.. include:: ../_shared/note-pip-install-determined.txt
+
+.. include:: ../_shared/note-local-dtrain-job.txt
+
+*****************************************************
+ Step 1: Get the Tutorial Files & Run the Experiment
+*****************************************************
+
+To follow along, you'll need the tutorial files.
+
+-  Create a new directory.
+
+-  Access the tutorial files via the :download:`core_api_pytorch_mnist.tgz
+   </examples/core_api_pytorch_mnist.tgz>` download link or directly from the `Github repository
+   <https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+For this step, we'll use our ``model_def.py`` script and its accompanying ``const.yaml`` experiment
+configuration file.
+
+-  From the directory containing our files, we'll begin by running the following command:
+
+.. code:: bash
+
+   det e create const.yaml . -f
+
+We don't have any data to plot yet, but we'll open the Determined WebUI to see that our experiment
+is running.
+
+-  To do this, we'll go to ``http://localhost:8080/``.
+-  To sign in, we'll accept the default determined username, leave the password empty, and then
+   click **Sign In**.
+-  In the WebUI, we'll select our experiment, and then go to the **Logs** tab.
+
+************************************
+ Step 2: Report Epoch-Based Metrics
+************************************
+
+In this section, we'll define our epoch metric.
+
+-  To follow along, use the ``model_def_metrics.py`` script and its accompanying ``metrics.yaml``
+   experiment configuration file.
+
+Our script, ``model_def_metrics.py``, is a modification of the ``model_def.py`` script. It already
+reports training and validation metrics to the Determined master and contains a ``steps_completed``
+variable that is needed to plot metrics on a graph in the WebUI.
+
+.. note::
+
+   For a full description of the Core API PyTorch MNIST Tutorial files, visit the
+   :ref:`core-getting-started`.
+
+Step 2.1: Modify the Train and Validation Methods
+=================================================
+
+Our script, ``model_def_metrics.py``, contains ``core_context.train``. This is used to report
+training and validation metrics.
+
+However, we also want to report epoch-based metrics and to allow Determined to keep track of the
+specific epoch for which training loss is being reported.
+
+-  To do this, we'll modify the train() method to include ``epoch_idx`` as a metric:
+
+.. code:: python
+
+   # Index by (batch_idx + 1) * (epoch-1) * len(train_loader)
+   # to continuously plot loss on one graph for consecutive
+   # epochs.
```

Comment:
Recommendation: drop this comment from the snippet.

As a developer, I usually read a comment first, because a comment indicates that the code is doing something non-obvious, or that it's a plain english explanation of what's going on.

But that comment is unhelpful, because it's drawing attention away from the line that was actually modified, which is the `metrics=` line.

Like, the comment makes sense from the perspective of somebody reading the original script, but it isn't the right focus from the perspective of this document.

Also, consider using `code:: diff` and using `-` in front of the line you're deleting and `+` in front of the line you are adding.  It will highlight the deleted line as red and the added line as green, in order to visually identify the modification to the script.  In fact, if you do that you could leave the comment if you'd like, because the dominant "focuser" would be the diff coloring, not the comment.

Reply:
dropped the comment, thanks for the tip!